### ISSUE TITLE
Exception encountered: Could not deserialize class 'Functional' because its parent module keras.src.engine.functional cannot be imported.

### ISSUE THREAD

**Issue by safer-lgtm** [safer-lgtm/leaf-disease-identification#1]
### Issue Description:

While working with TensorFlow 2.12.0, we encountered an issue with model deserialization when using custom objects. Specifically, when attempting to load a model saved as `model.keras`, we received the following error:

**`Exception encountered: Could not deserialize class 'Functional' because its parent module keras.src.engine.functional cannot be imported.`**

This issue is related to incompatibilities in how TensorFlow's Keras module handles custom layers and models, especially with the `.keras` serialization format.

**Comment by safer-lgtm** [comment]
### Steps Taken to Resolve:

1. **Installing Required Packages:**
   We installed the required dependencies to save models in the HDF5 format:

   ```bash
   pip install pyyaml h5py tensorflow==2.15.0
   ```

2. **Imports and Model Saving:**
   Initially, TensorFlow and Keras were imported as follows:

   ```python
   import tensorflow as tf
   from tensorflow import keras
   ```

3. **Optimizer Fix:**
   Instead of using the `Adam` optimizer with a learning rate parameter:

   ```python
   Adam(learning_rate=0.0001)
   ```

   We simplified it by directly specifying:

   ```python
   optimizer="adam"
   ```

4. **Switching to `.h5` Format:**
   Since the `.keras` format caused issues with model loading, we decided to save the model in the `.h5` format, which is more stable and better supported for model serialization in TensorFlow.

   ```python
   model.save("model.h5")
   ```

This solution should help resolve the deserialization error and provide a more stable workflow for model saving and loading in TensorFlow.

### Helpful References:

* [[TensorFlow Save and Load Tutorial](https://www.tensorflow.org/tutorials/keras/save_and_load#savedmodel_format)
* [[Keras GitHub Issue on Model Deserialization](https://github.com/keras-team/keras/issues/20167)
